docs/standard/serialization/system-text-json/nullable-annotations.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Respect nullable annotations
+description: "Learn how to configure serialization and deserialization to respect nullable annotations."
+ms.date: 10/22/2024
+no-loc: [System.Text.Json, Newtonsoft.Json]
+---
+# Respect nullable annotations
+
+Starting in .NET 9, <xref:System.Text.Json.JsonSerializer> has (limited) support for non-nullable reference type enforcement in both serialization and deserialization. You can toggle this support using the <xref:System.Text.Json.JsonSerializerOptions.RespectNullableAnnotations?displayProperty=nameWithType> flag.
+
+For example, the following code snippet throws a <xref:System.Text.Json.JsonException> during serialization with a message like:
+
+> The property or field 'Name' on type 'Person' doesn't allow getting null values. Consider updating its nullability annotation.
+
+:::code language="csharp" source="snippets/nullable-annotations/Nullable.cs" id="Serialization":::
+
+Similarly, <xref:System.Text.Json.JsonSerializerOptions.RespectNullableAnnotations> enforces nullability on deserialization. The following code snippet throws a <xref:System.Text.Json.JsonException> during serialization with a message like:
+
+> The constructor parameter 'Name' on type 'Person' doesn't allow null values. Consider updating its nullability annotation.
+
+:::code language="csharp" source="snippets/nullable-annotations/Nullable.cs" id="Deserialization":::
+
+> [!TIP]
```

Comment:
I neglected to put this in the blog, but would it be possible to add passing mention that the feature also recognizes the `[NotNull]`, `[AllowNull]`, `[MaybeNull]` and `[DisallowNull]` attributes? They're attributes used by the compiler to fine-tune annotations in getters and setters. I don't think we need examples, but I have a few tests highlighting the use case for your reference:

https://github.com/dotnet/runtime/blob/e16be4174212fd2abdfce2cfefb613f886264a54/src/libraries/System.Text.Json/tests/Common/NullableAnnotationsTests.cs#L560-L608

Reply:
If you choose to mention these attributes, you can link to [this](https://learn.microsoft.com/dotnet/csharp/language-reference/attributes/nullable-analysis) for the full docs on the attributes.